crates/uv-cli/src/lib.rs

```diff
@@ -832,6 +832,13 @@ pub struct PipSyncArgs {
 
     #[command(flatten)]
     pub compat_args: compat::PipSyncCompatArgs,
+
+    /// Allow sync of empty requirements, which will clear the environment of all packages.
+    #[arg(long, overrides_with("no_allow_empty_requirements"))]
+    pub allow_empty_requirements: bool,
```

Comment:
I would prefer `--allow-empty`, I don't think it's ambiguous.

I think this also needs to be added to the `PipOptions` and respected in persistent configuration, and it should _probably_ be accepted on `pip install` too?

Reply:
If `pip install` exists with success when the requirements are empty I don't think it would mean anything there.

I can shorten it, though I find it a bit ambiguous.

Reply:
I guess it makes sense to include in the persistent configuration too? It only seems relevant for this command though.

Reply:
Yeah it should be in the persistent configuration -- that's the contract for the `pip` table (e.g., it includes things that only apply to `pip-compile`).

Reply:
I find `--allow-empty-requirements` to be more verbose than necessary but not a strong opinion, it's fine.